packages/plugins/graphql-jit/src/index.ts

```diff
@@ -1,18 +1,81 @@
 /* eslint-disable no-console */
 import { Plugin } from '@envelop/types';
-import { GraphQLError } from 'graphql';
-import { compileQuery, isCompiledQuery, CompilerOptions } from 'graphql-jit';
+import { GraphQLError, DocumentNode, ExecutionResult, Source } from 'graphql';
+import { compileQuery, isCompiledQuery, CompilerOptions, CompiledQuery } from 'graphql-jit';
+import lru from 'tiny-lru';
+
+const DEFAULT_MAX = 1000;
+const DEFAULT_TTL = 3600000;
+
+export type JitCacheOptions = {
+  /**
+   * Maximum size of LRU Cache
+   * @default 1000
+   */
+  max?: number;
+  /**
+   * TTL in milliseconds
+   * @default 3600000
+   */
+  ttl?: number;
+};
 
 export const useGraphQlJit = (
   compilerOptions: Partial<CompilerOptions> = {},
-  pluginOptions: Partial<{
-    onError: (r: ReturnType<typeof compileQuery>) => void;
-  }> = {}
+  pluginOptions: {
+    onError?: (r: ReturnType<typeof compileQuery>) => void;
+  } & JitCacheOptions = {}
```

Comment:
Here I was just following the existing style in other plugins, but I think it should be better to create and use interfaces of all the options, and export them